### PR TITLE
Adding new distro support containers

### PR DESCRIPTION
**Pull request by AhmedBM**
## Description

* Adding new distro build containers (amd64) (retagged images requested for those missing)
  * Alma Linux 9
  * Amazon Linux 2
  * CentOS 7, 8
  * Oracle Linux 7, 8
  * RHEL 7, 8 ,9
  * Rocky Linux 9
  * SLES 15

## Checklist

- [X] I have read the [contribution guidelines](https://github.com/Azure/azure-osconfig/blob/main/CONTRIBUTING.md).
- [X] All unit tests are passing.
- [X] I have merged the latest `main` branch prior to this PR submission.
- [X] I submitted this PR against the `main` branch.

### CODE REVIEW

**Review by robertschaedler3 — APPROVED** [review]

File: devops/docker/almalinux-9-amd64/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# TODO: Add MSFT mirror
```
Comment (MariusNi) [inline devops/docker/almalinux-9-amd64/Dockerfile:1]: Sorry, what's this? We should not have "MSFT" in here. Applies everywhere else.

File: devops/docker/centos-8-amd64/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,37 @@
+# TODO: Add MSFT mirror
+FROM centos:latest
+
+# CentOS Linux 8 had reached the End Of Life (EOL) on December 31st, 2021. 
+# It means that CentOS 8 will no longer receive development resources from
+# the official CentOS project. After Dec 31st, 2021, if you need to update 
+# your CentOS, you need to change the mirrors to vault.centos.org where they
+# will be archived permanently
```
Comment (MariusNi) [inline devops/docker/centos-8-amd64/Dockerfile:8]: Until CentOS 8 will be removed from the list of officially targeted distros we need to have a package for it. This distro may have some serious user base.

File: devops/docker/rhel-7-amd64/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,42 @@
+# ubi7 images are missing RPM devtools, copy from centos7
+FROM centos:centos7 as centos
+RUN yum update -y && yumdownloader rpm-build && yumdownloader elfutils
+
+# TODO: Add MSFT mirror
```
Comment (MariusNi) [inline devops/docker/rhel-7-amd64/Dockerfile:5]: See the other comment. Applies everywhere. No "MSFT" in public GitHub repo, please.

File: devops/docker/almalinux-9-amd64/Dockerfile (hunk 1 above)
Comment (AhmedBM) [reply devops/docker/almalinux-9-amd64/Dockerfile:1]: sure, ill take it out

File: devops/docker/centos-8-amd64/Dockerfile (hunk 2 above)
Comment (AhmedBM) [reply devops/docker/centos-8-amd64/Dockerfile:8]: Agreed, i just placed the comment there to justify the changes from `mirror.centos.org` to `vault.centos.org`

File: devops/docker/rhel-7-amd64/Dockerfile (hunk 3 above)
Comment (AhmedBM) [reply devops/docker/rhel-7-amd64/Dockerfile]: addressed

File: devops/docker/rhel-8-amd64/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,30 @@
+# TODO: Add MSFT mirror
```
Comment (MariusNi) [inline devops/docker/rhel-8-amd64/Dockerfile:1]: Forgot one here

**Review by MariusNi — APPROVED** [review]
:shipit:

**Pull request merged**